doc/concepts.md

```diff
@@ -1,36 +1,55 @@
-Concepts
-========
-dlite shares the [metadata model of SOFT][1] and generic data stored
-with SOFT can be read with dlite and vice verse.  However, apart from
-dlite being much smaller and less complete, there are also a few
-notable differences in the API and even conceptual, which are detailed
-below.
+Relation to SOFT
+================
+SOFT stands for SINTEF Open Framework and Tools and is a set of
+concepts for how to achieve semantic interoperability as well as
+implementations and tooling.  At the core of SOFT are the [SOFT data
+models], which provide a by design simplistic but powerful way to
+represent scientific data.
+
+DLite is one implementation of the [SOFT data models].  Originally it
+was a simplified pure C implementation of SOFT5, but has with time
+developed into a project of its own.  This document will highlight a
+few notable differences with SOFT5.
 
 
 Named data instances
 --------------------
-In SOFT all Entity instances are referred to by their UUID.  However,
-in some cases when you have unique and immutable data, e.g. default
+In SOFT all instances are referred to by their UUID.  All metadata are
+uniquely identified by their URI, which must be of the form
+'`namespace`/`version`/`name`', like:
+http://onto-ns.com/meta/0.3/EntitySchema.  This is also valid in
+DLite, but we allow the URI to optionally end with a hash (#) or slash
+(/), which will be ignored.  Since a metadata is an instance of its
+meta-metadata, it also has an UUID, which is calculated from the URI
+(as an UUID version 5 SHA-1 hash of the URI using the DNS namespace).
+
+In some cases when you have unique and immutable data, e.g. default
 input parameters to a given version of a software model, it may be
 more convenient to refer to an unique human understandable name (URI),
-like "mymodel-1.2.3_default_input", rather than a UUID on the form
-"8290318f-258e-54e2-9838-bb187881f996".  dlite supports this.  The
-tool `dlite-getuuid` can be used to manually convert URIs to their
+like "`mymodel-1.2.3-default_input`", rather than a UUID on the form
+"`8290318f-258e-54e2-9838-bb187881f996`".  DLite supports this.
+Currently DLite does not enforce that user-defined URIs must follow the
+[RFC 3986] standard for a [valid URI], but it is recommended to
+follow it.
+
+The tool `dlite-getuuid` can be used to manually convert URIs to their
 corresponding UUIDs.
 
-If the `id` provided to dlite_datamodel() or dlite_instance_create()
-is not `NULL` or a valid UUID, it is interpreted as an unique uri
-referring to the instance.  dlite will then generate a version 5
-UUID from the `id` (from the SHA-1 hash of `id` and the DNS namespace)
-that the instance will be stored under.  The original `id` will also be
-stored and can be retrieved with dlite_datamodel_get_meta_uri().
+A resent feature in DLite is that data instances with no user-defined
```

Comment:
maybe not use resent, as the documentation in principle should last for a long time

Reply:
Documentation updated